src/front/glsl/context.rs

```diff
@@ -29,9 +29,9 @@ pub struct Context {
     pub lookup_global_var_exps: FastHashMap<String, VariableReference>,
     pub samplers: FastHashMap<Handle<Expression>, Handle<Expression>>,
 
-    pub hir_exprs: Arena<HirExpr>,
     pub typifier: Typifier,
     emitter: Emitter,
+    stmt_ctx: Option<StmtContext>,
```

Comment:
does this need to be an option? That prevents reusing the allocation of the arena.

Reply:
The `Option` is what allows the `Arena` to be reused, you can only have one `StmtContext` when you call `stmt_ctx` is takes if available from the `Context` and panics otherwise and later you call `lower` and it returns to the `Context`.

Reply:
oh I see, so this is to fight the borrow checker